hack/cleanup/konnect_control_planes.go

```diff
@@ -0,0 +1,133 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"net/url"
+	"strings"
+	"time"
+
+	sdkkonnectgo "github.com/Kong/sdk-konnect-go"
+	sdkkonnectcomp "github.com/Kong/sdk-konnect-go/models/components"
+	sdkkonnectops "github.com/Kong/sdk-konnect-go/models/operations"
+	"github.com/go-logr/logr"
+	"github.com/samber/lo"
+
+	"github.com/kong/kong-operator/controller/konnect/ops"
+	"github.com/kong/kong-operator/test"
+)
+
+const (
+	konnectControlPlanesLimit     = int64(100)
+	timeUntilControlPlaneOrphaned = time.Hour
+
+	kindKonnectGatewayControlPlane = "KonnectGatewayControlPlane"
+)
+
+// cleanupKonnectControlPlanes deletes orphaned control planes created by the tests and their roles.
+func cleanupKonnectControlPlanes(ctx context.Context, log logr.Logger) error {
+	serverURL, err := canonicalizedServerURL()
+	if err != nil {
+		return fmt.Errorf("invalid server URL %s: %w", test.KonnectServerURL(), err)
+	}
+
```

Comment:
REVIEW: We removed the check and delete of the roles here. Do we still need them?

Reply:
~AFAIR this has been fixed in Konnect and roles are deleted on CP deletion, that was the reason for removing this.~

I can still see KIC's job handling role deletion https://github.com/Kong/kubernetes-ingress-controller/blame/c29a8c84afdb1d6a04b93789b6a0faf8f9cf24e3/hack/cleanup/konnect_control_planes.go#L51-L58 so I believe we need to do that as well here.